FEXCore/Source/Interface/Config/Config.cpp

```diff
@@ -318,65 +316,61 @@ fextl::string FindContainerPrefix() {
 void ReloadMetaLayer() {
   Meta->Load();
 
-  if (FEXCore::Config::Exists(FEXCore::Config::CONFIG_CACHEOBJECTCODECOMPILATION)) {
-    FEX_CONFIG_OPT(CacheObjectCodeCompilation, CACHEOBJECTCODECOMPILATION);
-  }
-
-  fextl::string ContainerPrefix {FindContainerPrefix()};
-  auto ExpandPathIfExists = [&ContainerPrefix](FEXCore::Config::ConfigOption Config, fextl::string PathName) {
-    auto NewPath = ExpandPath(ContainerPrefix, PathName);
+  const fextl::string ContainerPrefix {FindContainerPrefix()};
+  auto ExpandPathIfExists = [&ContainerPrefix](FEXCore::Config::ConfigOption Config, const fextl::string* PathName) {
```

Comment:
Turning `PathName` into a pointer doesn't help much here AFAICT. We never reused the external storage, and passing in rvalues (as we've done before) will avoid unnecessary copies already.

Reply:
This isn't where the copy was occuring, the copy was occurring below in `FEX_CONFIG_OPT`, this is just to change the interface to match what `Meta->Get` returns which is a `std::optional<std::string*>` which is a pointer to the internal string tracked by the config system.
This just lets us pass that as a const pointer all the way through to `ExpandPathIfExists`->`ExpandPath` so it's matching the interface.

Reply:
Oof, I see. Can we at least use a const reference to highlight that these pointers won't randomly be null? This seems to be the assumption anyway since they're immediately dereferenced in the function.

Reply:
Sure, switched over to a const ref